packages/compass-schema/src/components/schema-toolbar/schema-toolbar.spec.tsx

```diff
@@ -0,0 +1,149 @@
+import userEvent from '@testing-library/user-event';
+
+import { SchemaToolbar } from './schema-toolbar';
+import AppRegistry from 'hadron-app-registry';
+
+const mockQueryBarRole = {
+  name: 'Query Bar',
+  // eslint-disable-next-line react/display-name
+  component: () => <div>Query bar</div>,
+  configureStore: () => ({}),
+  configureActions: () => {},
+  storeName: 'Query.Store',
+  actionName: 'Query.Actions',
+};
+
+const mockQueryBarStore = {
+  state: {
+    filterString: '123',
+    projectString: '',
+    sortString: '',
+    collationString: '',
+    skipString: '',
+    limitString: '',
+    maxTimeMSString: '',
+  },
+};
+
+const renderSchemaToolbar = (
+  props: Partial<ComponentProps<typeof SchemaToolbar>> = {}
+) => {
+  const localAppRegistry = new AppRegistry();
+  localAppRegistry.registerRole('Query.QueryBar', mockQueryBarRole);
+  localAppRegistry.registerStore(mockQueryBarRole.storeName, mockQueryBarStore);
+
+  render(
+    <SchemaToolbar
+      onExportToLanguageClicked={() => {}}
+      localAppRegistry={localAppRegistry}
+      analysisState="complete"
+      errorMessage={''}
+      isOutdated={false}
+      onAnalyzeSchemaClicked={() => {}}
+      onResetClicked={() => {}}
+      sampleSize={10}
+      schemaResultId="123"
+      {...props}
+    />
+  );
+};
+
+describe('SchemaToolbar', function () {
+  afterEach(function () {
+    sinon.restore();
+  });
+
+  it('calls onExportToLanguage with the query state when the export to language is clicked', function () {
+    const localAppRegistry = new AppRegistry();
+    const onExportToLanguageClickedSpy = sinon.spy();
+
+    localAppRegistry.registerRole('Query.QueryBar', mockQueryBarRole);
+    localAppRegistry.registerStore(
+      mockQueryBarRole.storeName,
+      mockQueryBarStore
+    );
+
+    renderSchemaToolbar({
+      onExportToLanguageClicked: onExportToLanguageClickedSpy,
+      localAppRegistry,
+    });
+
+    expect(onExportToLanguageClickedSpy.called).to.be.false;
+    userEvent.click(screen.getByRole('button'));
+
+    expect(onExportToLanguageClickedSpy.calledOnce).to.be.true;
+    expect(onExportToLanguageClickedSpy.firstCall.args[0]).to.deep.equal({
+      filterString: '123',
+      projectString: '',
+      sortString: '',
+      collationString: '',
+      skipString: '',
+      limitString: '',
+      maxTimeMSString: '',
+    });
+  });
+
+  it("renders errors when they're passed", function () {
+    renderSchemaToolbar({
+      analysisState: 'error',
+      errorMessage: 'test error msg',
+    });
+
+    expect(
+      screen.getByText(
+        'An error occurred during schema analysis: test error msg'
+      )
+    ).to.be.visible;
+  });
+
+  it('does not render errors when the analysis state is not error', function () {
+    renderSchemaToolbar({
+      errorMessage: 'test error msg',
+    });
+
+    expect(
+      screen.queryByText(
+        'An error occurred during schema analysis: test error msg'
+      )
+    ).to.not.exist;
```

Comment:
The thing about these assertions is that any typo in the text above would also cause it to not exist.

Reply:
Updated it to also test with a `data-testid` so it's not all dependent on the text, and updated it to use a shared variable for the text, I think that'll make it a bit more resilient, thanks for the suggestion. Unit tests like these tests are fairly brittle. I usually try to keep tests independent of variables used in the actual code so ideally break / need to be updated when we change some text or rendering. Additionally I like to keep tests as non-coupled as possible so teardown and changes are simpler.

I'm not sure I follow the other suggestion, what would be the action?